cybersecuritytools/csls/generate_cloudwatch_logs/cli.py

```diff
@@ -0,0 +1,129 @@
+@click.option("--ssm", required=True, help="SSM root path")
+def smoke_test(ssm_root: str, timeout: int) -> None:
+    """Run an end to end test on the pipeline"""
+    cloudwatch_results = send_logs_to_cloudwatch()
+    print("Sent logs to CloudWatch")
+    start_timestamp = int(datetime.now().timestamp())
+    print("Polling splunk to find our logs...")
+
+    splunk_credentials = credentials(ssm_root, "search")
+    splunk = Search(splunk_credentials)
+
+    while True:
+        duration = int(datetime.now().timestamp()) - start_timestamp
+        splunk_results = splunk.search(search_query(test_type="smoke_test"))
+
+        if payload_found(cloudwatch_results, splunk_results):
+            print(f"\n✔️ Pipeline smoketest succeeded in {duration} seconds")
+            sys.exit(0)
+
+        if duration > timeout:
+            print(
+                f"\n❌TIMEOUT searching for payload in splunk after {duration} seconds",
+                file=sys.stderr,
+            )
+            print("CloudWatch results: ")
+            pprint(cloudwatch_results)
+            print("\n\n\n\n")
+            print("Splunk results: ")
+            pprint(splunk_results)
+            print(
+                f"\n❌ TIMEOUT searching for payload in splunk after {duration} seconds",
+                file=sys.stderr,
+            )
+            sys.exit(1)
+
+        sleep(1)
+        print(".", end="", flush=True)
+
+
+@generate_cloudwatch_logs.command()
+@click.option("-t", "--timeout", type=int, default=600)
+@click.option("--ssm", required=True, help="SSM root path")
+def load_test(ssm: str, timeout: int) -> None:
+    """Check Splunk for artillery payloads on the pipeline"""
+    start_timestamp = float(datetime.now().timestamp())
+
+    splunk_credentials = credentials(ssm, "search")
+    splunk = Search(splunk_credentials)
+
+    while True:
+
+        duration = float(datetime.now().timestamp()) - start_timestamp
+        artillery_config = 80000
+        requests_completed = int(os.environ.get("requests_completed", artillery_config))
+
+        splunk_results = splunk.search(search_query(test_type="smoke_test"))
+
+        if load_test_found(splunk_results, artillery_config, requests_completed):
+            print(f"\n✔️ Pipeline load test succeeded in {duration} seconds")
+            sys.exit(0)
+
+        if requests_completed < int(artillery_config * 0.9):
+            print(
+                f"\n❌INSUFFICIENT DATA Artillery has only sent {requests_completed}"
+                f"requests from a config of {artillery_config}"
+            )
+            sys.exit(0)
+
+        if duration > timeout:
+            print(
+                f"\n❌TIMEOUT searching for payload in splunk after {duration} seconds",
+                file=sys.stderr,
+            )
+            print("Splunk results: ")
+            pprint(splunk_results)
+            print(
+                f"\n❌ TIMEOUT searching for payload in splunk after {duration} seconds",
```

Comment:
I hate to be THAT GUY, but there's inconsistent spacing after the emoji. This can sometimes be caused by editor support.

Reply:
Please see: https://github.com/alphagov/centralised-security-logging-service/commit/905403cfcd1e666e45a6f98d2671caa026ac47ed#diff-52061f2ecc5d3b116eff0c03326d9595R68